File: apps/api/src/auth/commands/local-sign-in.handler.ts
import { Logger } from '@nestjs/common';
import { CommandHandler, EventPublisher, ICommandHandler } from '@nestjs/cqrs';
import { LocalSignInCommand } from '@/auth/commands/local-sign-in.command';

@CommandHandler(LocalSignInCommand)
export class LocalSignInHandler implements ICommandHandler<LocalSignInCommand> {
  private readonly logger = new Logger(LocalSignInHandler.name);

  constructor(private readonly publisher: EventPublisher) {}

  async execute(command: LocalSignInCommand) {}
}

File: apps/cms/src/Modules/Auth/Handlers/LocalSignInHandler.cs
using MediatR;
using GameGuild.Modules.Auth.Commands;
using GameGuild.Modules.Auth.Dtos;
using GameGuild.Modules.Auth.Services;

namespace GameGuild.Modules.Auth.Handlers;

/// <summary>
/// Handler for local sign-in command
/// </summary>
public class LocalSignInHandler : IRequestHandler<LocalSignInCommand, SignInResponseDto>
{
    private readonly IAuthService _authService;

    public LocalSignInHandler(IAuthService authService)
    {
        _authService = authService;
    }

    public async Task<SignInResponseDto> Handle(LocalSignInCommand request, CancellationToken cancellationToken)
    {
        var signInRequest = new LocalSignInRequestDto
        {
            Email = request.Email,
            Password = request.Password
        };

        return await _authService.LocalSignInAsync(signInRequest);
    }
}
